src/main/java/org/kaazing/netx/ws/internal/io/OutgoingSentinelExtension.java

```diff
@@ -262,7 +262,7 @@ else if (reasonLength > 123) {
 
             int reasonOffset = closePayload.reasonOffset();
 
-            connection.getRandom().nextBytes(mask);
+            createMask(connection, mask);
```

Comment:
We need to keep in a separate so that we can xor(^) the payload bytes appropriately.